boxsdk/object/cloneable.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+
+class Cloneable(object):
+    """
+    Cloneable interface to be implemented by endpoint objects that should have ability to be cloned, but with a
+    different session member if desired.
+    """
+
+    def as_user(self, user):
+        """
+        Returns a new endpoint object with default headers set up to make requests as the specified user.
+        :param user:
+            The user to impersonate when making API requests.
+        :type user:
+            :class:`User`
+        """
+        return self.clone(self.session.as_user(user))
+
+    def with_shared_link(self, shared_link, shared_link_password):
+        """
+        Returns a new endpoint object with default headers set up to make requests using the shared link for auth.
```

Comment:
Sphinx needs blank lines between the title, and the rest of the docstring.

This is fine for now, but we should probably do a test run of sphinx, and fix all the errors, before doing a release.